src/index.ts

```diff
@@ -1,36 +1,58 @@
 import server from 'server'
 import { status, render, redirect, Reply } from 'server/reply'
 import { get, post } from 'server/router'
-import fundAccount, { FUNDABLE_AMOUNT } from './fundAccount'
 
-const rateLimiter: { [key: string]: Date } = {}
+import { Connection } from './shared/connections'
+import fundAccount from './fundAccount'
+
+const rateLimiter: {
+  [currency: string]: {
+    [key: string]: Date
+  }
+} = {}
 const limit = 1000 * 60 * 60 // 1 hour
 
 server({ port: 3000, public: 'dist', views: 'dist', security: false }, [
-  get('/', _ctx => render('index.html')),
+  get('/', _ => render('index.html')),
   get('/*', _ => redirect('/')),
-  post('/fund/:address', ctx => handleFundRequest(ctx.params.address))
+  post('/fund/:address', ctx => {
+    const connection: Connection = JSON.parse(ctx.data)
+    return handleFundRequest(ctx.params.address, connection)
+  }),
 ])
 
-async function handleFundRequest(address: string): Promise<Reply> {
+async function handleFundRequest(address: string, connection: Connection): Promise<Reply> {
   const requestTime = new Date()
-  if (isOverRateLimit(address, requestTime)) {
-    return status(429).json({ limitEnd: rateLimiter[address] })
+  const { currency } = connection
+
+  if (isOverRateLimit(address, currency, requestTime)) {
+    return status(429).json({ limitEnd: rateLimiter[currency][address] })
```

Comment:
`rateLimiter` is defined outside the function scope. It might be better to have it defined inside `handleFundRequest` to avoid a reference to something outside.

Reply:
rateLimiter is meant to be a module level variable.